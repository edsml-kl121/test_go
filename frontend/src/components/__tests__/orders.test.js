import {render, screen, cleanup, getByTestId} from "@testing-library/react"
import Orders from "../orders.components";
import renderer from 'react-test-renderer'

afterEach(() => {
  cleanup()
})

test('should render todo component', () => {
  render(<Orders/>)
  const orderElement = screen.getByTestId('orders-1')
  expect(orderElement).toBeInTheDocument();
  expect(orderElement).toHaveTextContent('The Order Adder App');
})

test('matches snapshot', () => {
  const tree = renderer.create(<Orders/>)
  expect(tree).toMatchSnapshot();
})

test('buttons are disabled', () => {
  render(<Orders/>)
  const button = screen.getByTestId('add-new-order')
  expect(button).not.toBeDisabled();
  // expect(screen.getByRole({ name: "Add new order" })).toBeDisabled();
  // expect(Orders(/Add new order/i)).toBeDisabled()
})
