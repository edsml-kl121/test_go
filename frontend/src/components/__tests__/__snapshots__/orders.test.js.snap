// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    data-testid="orders-1"
  >
    The Order Adder App
  </div>
  <div
    className="container"
  >
    <button
      className="btn btn-primary"
      data-testid="add-new-order"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Add new order
    </button>
  </div>
  <div
    className="container"
  />
</div>
`;
